src/providers/drafts.ts

```diff
@@ -0,0 +1,140 @@
+import { Language, License, VideoCategory } from '@/api/queries'
+import { createStore } from '@/store'
+import { createId } from '@/utils/createId'
+import { readFromLocalStorage } from '@/utils/localStorage'
+
+export type CommonDraftProps = {
+  id: string
+  channelId: string
+  updatedAt: string
+  seen?: boolean
+}
+
+export type Draft = VideoDraft
+
+export type DraftType = 'video'
+
+export type VideoDraft = {
+  type: 'video'
+  title?: string | null
+  description?: string | null
+  isPublic?: boolean | null
+  publishedBeforeJoystream?: string | null
+  hasMarketing?: boolean | null
+  isExplicit?: boolean | null
+  language?: Language['iso'] | null
+  category?: VideoCategory['id'] | null
+  licenseCode?: License['code'] | null
+  licenseCustomText?: License['customText'] | null
+  licenseAttribution?: License['attribution'] | null
+} & CommonDraftProps
+
+export type RawDraft = Omit<Draft, 'id' | 'updatedAt'>
+interface DraftStoreStateV1 {
+  allDrafts: Draft[]
+  allUnseenDrafts: Array<{
+    draftId: string
+    channelId: string
+  }>
+}
+interface DraftStoreStateV2 {
+  allDrafts: Draft[]
+}
+interface DraftStoreActions {
+  addDraft: (draft: RawDraft, explicitId?: string) => Draft
+  updateDraft: (draftId: string, draftProps: RawDraft) => void
+  getDraft: (draftId: string) => Draft | undefined
+  getDraftsForChannel: (draftIds: string) => Draft[]
+  getUnseenDraftsForChannel: (draftIds: string) => Draft[]
+  markAllDraftsAsSeenForChannel: (channelId: string) => void
+  removeDrafts: (draftIds: string[]) => void
+  removeAllDrafts: (channelId: string) => void
+}
+
+export const useDraftStore = createStore<DraftStoreStateV2, DraftStoreActions>(
+  {
+    state: {
+      allDrafts: [], // includes drafts for different channels
+    },
+    actionsFactory: (set, get) => ({
+      addDraft: (draft, explicitId) => {
+        const id = explicitId ?? createId()
+        const updatedAt = new Date().toISOString()
+        const newDraft: Draft = { ...draft, updatedAt, id, seen: false }
+        set((draftState) => {
+          draftState.allDrafts = [newDraft, ...draftState.allDrafts]
+        })
+        return newDraft
+      },
+      updateDraft: (draftId, draftProps) => {
+        const updatedAt = new Date().toISOString()
+        set((draftState) => {
+          const idx = draftState.allDrafts.findIndex((d) => d.id === draftId)
+          if (idx >= 0) {
+            draftState.allDrafts[idx] = { ...draftState.allDrafts[idx], ...draftProps, updatedAt }
+          }
+        })
+      },
+      getDraft: (id) => get().allDrafts.find((draft) => draft.id === id),
+      getDraftsForChannel: (channelId) => get().allDrafts.filter((d) => d.channelId === channelId),
+      getUnseenDraftsForChannel: (channelId) =>
+        get().allDrafts.filter((d) => d.channelId === channelId && d.seen === false),
+      removeDrafts: (draftIds) => {
```

Comment:
Why did we change to remove an array of drafts? I think we always use this to delete only one draft?

Reply:
Before we would accept a draftId or an array of draftIds and we had to maintain both scenarios and check if draftIds was an array or not; so I thought this way we only had to maintain one implementation and it wouldn't affect usage besides enclosing draftId in an array if removing only one.

Right now, there is not much use case for multiple removals so use case unclear, but before there was when we had the checkboxes, and possibly in the future there will be (?) seems worth having the option to remove multiple drafts without iterating an array but 🤷‍♂